wallet/module/wallet.go

```diff
@@ -92,6 +95,13 @@ func (m *Module) NewAddress(ctx context.Context, typ string) (string, error) {
 	}
 
 	if m.masterAddr != address.Undef {
+		balance, err := client.WalletBalance(ctx, m.masterAddr)
+		if err != nil {
+			return "", fmt.Errorf("getting balance from master addr: %s", err)
+		}
+		if balance.LessThan(types.BigAdd(types.BigInt{Int: m.iAmount}, types.NewInt(feeThreshold))) {
```

Comment:
Fail fast if most prob the txn will fail if we don't have enough funds. The txn is sent async since is a blocking process in the Lotus API now since has to estimate feeds and that takes time.

We still have two improvements to do here:
- Instead of using `feeThreshold` as a constant, we could craft a more dynamic calculation of estimated fees. (#610)
- Think about a possibly more reliable way to send the funding txn. If for some wild reason the sent txn failed, we'd get some unwanted result of a 0 balance addr. So looks like we should have a better tracking mechanism about this... not sure though what might be a good solution apart from just retrying with some exponential backoff. (#611)